yt/frontends/gadget/io.py

```diff
@@ -223,6 +271,9 @@ def _identify_fields(self, data_file):
                         self._vector_fields[kk] = g[kk].shape[1]
                     fields.append((ptype, str(kk)))
 
+        if self.ds.gen_hsmls:
```

Comment:
This should be _after_ the `f.close()` line (better closing a file as soon as we're done with io operations)

Reply:
Agreed.